blockchain/projects.go

```diff
@@ -21,13 +21,10 @@ package blockchain
 import (
 	"encoding/json"
```

Comment:
just changes from extracting cache and mutex implementation

Reply:
and renaming projects receiver from `s` -> `p` (it was s from before when it was called state)